code/game/machinery/kitchen/gibber.dm

```diff
@@ -207,8 +207,14 @@ obj/machinery/gibber/New()
 
 	var/obj/item/weapon/reagent_containers/food/snacks/meat/allmeat[totalslabs]
 	for (var/i=1 to totalslabs)
-		var/obj/item/weapon/reagent_containers/food/snacks/meat/newmeat = new occupant.meat_type(null, occupant)
-		newmeat.reagents.add_reagent (NUTRIMENT, sourcenutriment / totalslabs) // Thehehe. Fat guys go first
+		var/obj/item/weapon/newmeat
+		if(istype(occupant.meat_type, /obj/item/weapon/reagent_containers))
+			newmeat = new occupant.meat_type(null, occupant)
+		else
+			newmeat = new occupant.meat_type()
+
+		if(istype(newmeat, /obj/item/weapon/reagent_containers))
+			newmeat.reagents.add_reagent (NUTRIMENT, sourcenutriment / totalslabs) // Thehehe. Fat guys go first
```

Comment:
move this after `newmeat = new occupant.meat_type(null, occupant)`, you're already typechecking there

Reply:
Good call.